superset/assets/src/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -155,7 +158,7 @@ class SqlEditorLeftBar extends React.PureComponent {
             dataEndpoint={
               '/databaseasync/api/' +
               'read?_flt_0_expose_in_sqllab=1&' +
-              '_oc_DatabaseAsync=database_name&' +
+              '_oc_DatabaseAsync=verbose_name&' +
```

Comment:
Doesn't this mess up the ordering?

Reply:
Hm, yeah. However, if verbose names are displayed while ordering by database name, then it could potentially be just as confusing. Not sure how to address this.

Reply:
I'm also unclear on how to make this work. If it was sql you'd ORDER BY `COALESCE(verbose_name, database_name)` but being the FAB url mini-language I can't think of a way to do this without changing FAB

Reply:
What if I were to do the "coalescing" when creating/updating the row from the form? For example: 
- Add a checkbox to the form next to verbose name saying something like "use database name" and have it checked by default. It would simply copy the value of the database name field.
- If the user specifies any name that doesn't match the database name, the checkbox unchecks and their value is persisted on save.

Reply:
I think you could simply:
```
if not self.verbose_name:
    self.verbose_name = self.name
```
But then you also need a db migration that does this on top of the on_add and on_update events in the model view.

You could also:
* create a REST endpoint that does the sorting (not using FAB's api and using custom endpoint instsead)
* add support to property sorting in FAB (that's probably hard)

Reply:
Good call. I chose the new REST endpoint approach. Seemed like the easiest one to implement.